File: Source/Ivxr.SpaceEngineers/WorldModel/SessionInfo.cs
namespace Iv4xr.SpaceEngineers.WorldModel
{
    public class SessionInfo
    {
        public string Name;
        public string Description;
        public string CurrentPath;
        public bool IsAdminMenuEnabled;
        public bool IsRunningExperimental;
        public bool Ready;
        public bool IsUnloading;
        public bool IsCopyPastingEnabled;
        public bool StreamingInProgress;
        public bool IsServer;
        public bool IsPausable;
        public DefinitionId GameDefinition;
        public SessionSettings Settings;
        public CameraController Camera;
    }

    public class CameraController
    {
        public bool IsInFirstPersonView;
        public bool ForceFirstPersonCamera;
        public Entity Entity;
        public CameraControllerEnum CameraControllerEnum;
    }

    public enum CameraControllerEnum
    {
        Spectator,
        Entity,
        ThirdPersonSpectator,
        SpectatorDelta,
        SpectatorFixed,
        SpectatorOrbit,
        SpectatorFreeMouse
    }

    public class SessionSettings
    {
        public GameModeEnum GameMode;
        public bool InfiniteAmmo;
    }

    public enum GameModeEnum
    {
        Creative,
        Survival,
    }
}

File: JvmClient/src/commonMain/kotlin/spaceEngineers/model/SessionInfo.kt
package spaceEngineers.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class CameraController(
    @SerialName("IsInFirstPersonView")
    val isInFirstPersonView: Boolean,
    @SerialName("ForceFirstPersonCamera")
    val forceFirstPersonCamera: Boolean,
    @SerialName("CameraControllerEnum")
    val cameraControllerEnum: Int
)

@Serializable
data class SessionInfo(
    @SerialName("Name")
    val name: String,
    @SerialName("Description")
    val description: String? = null,
    @SerialName("CurrentPath")
    val currentPath: String,
    @SerialName("IsAdminMenuEnabled")
    val isAdminMenuEnabled: Boolean,
    @SerialName("IsRunningExperimental")
    val isRunningExperimental: Boolean,
    @SerialName("Ready")
    val ready: Boolean,
    @SerialName("IsUnloading")
    val isUnloading: Boolean,
    @SerialName("IsCopyPastingEnabled")
    val isCopyPastingEnabled: Boolean,
    @SerialName("StreamingInProgress")
    val streamingInProgress: Boolean,
    @SerialName("IsServer")
    val isServer: Boolean,
    @SerialName("IsPausable")
    val isPausable: Boolean,
    @SerialName("GameDefinition")
    val gameDefinition: DefinitionId,
    @SerialName("Settings")
    val settings: SessionSettings,
    @SerialName("Camera")
    val camera: CameraController? = null,
)
